pkg/consensus/payload.go

```diff
@@ -167,10 +182,7 @@ func (p *Payload) EncodeBinaryUnsigned(w *io.BinWriter) {
 	w.WriteU32LE(p.height)
 	w.WriteU16LE(p.validatorIndex)
 	w.WriteU32LE(p.timestamp)
-
-	ww := io.NewBufBinWriter()
-	p.message.EncodeBinary(ww.BinWriter)
-	w.WriteVarBytes(ww.Bytes())
+	w.WriteVarBytes(p.data)
```

Comment:
Maybe something like
```go
if p.message != nil {
    ww := io.NewBufBinWriter()
    p.message.EncodeBinary(ww.BinWriter)
    p.data = ww.Bytes()
}
w.WriteVarBytes(p.data)
```
And then you don't need to update `p.data` in every `Set` method.